sdk/opendp/whitenoise/client/__init__.py

```diff
@@ -34,24 +52,71 @@ def submit(self, params, uri):
         return client.executerun(details)
 
 class DatasetClient(object):
-    def register(self, dataset):
+    """
+    A client for registering, reading and releasing differentially private
+    datasets using the opendp-whitenoise service
+    """
+    def __init__(self):
+        # Tag requests with this custom header for now
+        self._guid = _guid_header()
+
+    def readreleased(self, dataset_name, budget):
+        """
+        Generates a DatasetReadReleaseRequest and sends it to the service.
+        Tags the request with Client guid.
+        """
         client = _get_client()
-
+        read_released_request = DatasetReadReleaseRequest(dataset_name=dataset_name, budget=budget)
+        return client.datasetreadreleased(read_released_request, custom_headers={'client_guid': self._guid})
+
+    def _register_release_request_helper(self, dataset):
+        """
+        Helper for register/release, 
+        both of which use DatasetDocuments as request formats
+        """
         for key in KNOWN_DATASET_TYPE_KEYS:
             if not key in dataset:
                 dataset[key]=None
 
-        register_request = DatasetDocument(dataset_name=dataset['dataset_name'], \
+        request = DatasetDocument(dataset_name=dataset['dataset_name'], \
             dataset_type=dataset['dataset_type'], \
             budget=dataset['budget'], \
             csv_details=dataset['csv_details'], \
-            dataverse_details=dataset['dataverse_details'])
-        return client.datasetregister(register_request)
+            dataverse_details=dataset['dataverse_details'], \
+            authorized_users=dataset['authorized_users'])
+        
+        return request
+
+    def release(self, dataset):
```

Comment:
the dataset document should include a budget param that is the cost of creating that release

Reply:
I added a new field called "release_cost," although should there be bothe a general "budget" field and a "release_cost" field? (both are currently in the spec)

Reply:
We can leave them in one doc for now.